### PR TITLE
polygon2d editing using transparent overlay

### PR DESCRIPTION
**Pull request by ghost**
This is an alternative version of #10614 based on @RandomShaper's proposal of using transparent previews. I'm not sure if I got his idea right.

Here's what it looks like currently as far as I understood it:

poly-transparent

Personally, I'd still prefer the original #10614 as a default/configurable default for two reasons:

(1) I don't personally see the use case where you really need to compare the new shape to the old shape in this way. To me it feels more like a technical gimmick that adds visual complexity to the editing (due to the transparency) without giving a real advantage.

(2) every pro design/vector app for people who are doing this kind of stuff all day long with really complex illustrations and vector shapes seems to change the shape as you edit it (they all skipped the preview lasso thing they had like 10 years ago due to technical limits of not being able to preview the changed shape in real time):

e.g. Inkscape:

inkscape

e.g. Affinity Designer:

affinity-designer

### CODE REVIEW

**Comment by toger5** [comment]
I really like this one. I see your point that it is not used in any other tool. But honestly i think it is a missing feature in other tools.

If it is easy, i would change the colors.
The old shape gets greyed out with transperency and the new shape is solid. Maybe i even wpuld draw the new shape above?

**Comment by Noshyaar** [comment]
Like toger5 says, I think RandomShaper's idea is that the new polygon filled (same as #10614 and other applications) and the old polygon outlined and faded. (Basically this PR but with colors switched.)

**Comment by poke1024** [comment]
OK, I see, swapped now. The old polygon is transparent now, the new one is solid. It's now drawing the new shape above (which I like, but the problem is obviously that it can hide the old shape now). If we want to draw the old shape above, then transparency alone is not enough, and indeed has to be some additional darkening/hue shift.

polygon-editing

**Comment by Noshyaar** [comment]
From my interpretation of his idea, I think it's also that old polygon is outlined only (and not filled) and is drawn on top of the new shape, probably.

**Comment by toger5** [comment]
The outline might not even be necassary. Although it would be cool.
I think the biggest use case is, if i want to check the old position. But aince i can just drag it back to see the old shape i always can get the old posution as reference.

**Comment by poke1024** [comment]
Here's three more experiments:

With an outline of the old polygon on top (as suggested by @Noshyaar):

outline

With additional transparency for the old polygon (which is basically this original PR with swapped old and new):

transparent

With additional color coding as originally suggested by @RandomShaper:

color

I find neither visually attractive and all make little sense to me personally. Frankly, I find them utterly confusing visually. I like the online only variant best as it's less intrusive. The code is quite complex for the transparent polygon overlay. I've amended this PR to the first, outline-only variant.

**Comment by akien-mga** [comment]
The first variant with only the outline is also what I had in mind, IMO it looks cool. If you really think it's a bother, feel free to add an editor option to disable it, but leave it enabled by default for discoverability.

**Comment by toger5** [comment]
I also lile the first one the most.
Less flashy when startingto drag a vertex.

**Comment by toger5** [comment]
@groud I think it helps in giving you information how much you actually moved the vertex.
If you move it slowly you don't how much you actually moved and changed the shape.
with the outline it's really easy to tell.

@poke1024 Thank you for adapting and giving so many implementation options!

**Comment by akien-mga** [comment]
> I think it helps in giving you information how much you actually moved the vertex.
If you move it slowly you don't how much you actually moved and changed the shape.
with the outline it's really easy to tell.

Yes, especially when adjusting collision shapes based on play testing, you might know how the previous shape behaves and what was off, and want to just tweak it a bit visually. Seeing the outline in such case helps know how it was, to establish how you want it to become.

**Comment by poke1024** [comment]
OK, thanks for the feedback. I'm closing this and continue with #10614 again, where I push the outline version.

**Comment by RandomShaper** [comment]
TL;DR I meant do draw only the outline. The translucent filled poly adds too much clutter indeed.
